### DISCUSSION TITLE
SelectButton in PrimeVue 4.x can always be deselected.

### DISCUSSION THREAD

**Discussion by BrutalSoloH** [primefaces/community#3099] [PrimeVue]
I have this code:
```
<SelectButton v-model="config.cardOrder" :options="optCardOrder" option-label="label"
                                    option-value="value" :allow-empty="false" />
```
But still I can deselect all options while allowEmpty = false. What am I doing wrong?

**Comment by j0rgedev** [comment]
Hey @BrutalSoloH could you share the config.cardOrder value?

**Comment by BrutalSoloH** [comment]
It's defined like this..

```ts
export enum CardOrder {
    alternating = 1,
    ranksuit = 2,
};

export class Config {
    cardOrder = CardOrder.alternating;
};

const config = reactive(new Config()) as Config;

app.provide('config', config)

....

const config = inject<Config>("config")!
const optCardOrder = computed(() => [{
    label: "Alternating",
    value: CardOrder.alternating
}, {
    label: "By Rank,
    value: CardOrder.ranksuit
}]);
.... 

<SelectButton v-model="config.cardOrder" :options="optCardOrder" option-label="label"
                                    option-value="value" :allow-empty="false" />


```

If I deselect the option (so nothing is selected anymore) then the value of config.cardOrder remains the value of the last selected option.
What I can see from the vue inspector is that the underlying ToggleButton's d-value gets set to false, but it's modelValue remains true.

**Comment by j0rgedev** [comment]
![image](https://github.com/user-attachments/assets/8c9e59ed-c156-4b5e-9c02-1b78c7a1a3ca)
Isn't that the correct behavior? If allow-empty is false, then the card order would never be null, and it'll remaing the last selected value. What value are you expecting?

**Comment by BrutalSoloH** [comment]
The value is expected, but visually none of the buttons are selected, so the user doesn’t know what value is selected. 
Example: when the left button is down (selected) and you click on it, it will become up or deselected. The value remains the same.
